File: cash.c
// Implementation of Greedy algorithm
// Program allows the user to input how much change is owed
// Then prints the minimum number of coins with which that change can be made

#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    int cents_owed;

    do
    {
        // Prompts user for valid input of amount of change owed
        float dollars_owed = get_float("Change owed: ");
        cents_owed = round(dollars_owed * 100);
    }
    while (cents_owed < 0);

    //Finding the best immediate change
    int quarters = cents_owed / 25;
    int dimes = (cents_owed % 25) / 10;
    int nickels = ((cents_owed % 25) % 10) / 5;
    int pennies = ((cents_owed % 25) % 10) % 5;

    printf("%d\n", quarters + dimes + nickels + pennies);
}

File: sentimental/cash.py
# Implementation of Greedy algorithm
# User inputs how much change is owed
# Program prints the minimum number of coins w/ which that change can be made

from cs50 import get_float


def main():
    # Prompts user for valid input of amount of change owed
    while True:
        dollars_owed = get_float("Change owed: ")
        cents_owed = round(dollars_owed * 100)
        if cents_owed > 0:
            break

    # Finding the best immediate change
    quarters = cents_owed // 25
    dimes = (cents_owed % 25) // 10
    nickels = ((cents_owed % 25) % 10) // 5
    pennies = ((cents_owed % 25) % 10) % 5

    # Prints minimum number of coins
    print(quarters + dimes + nickels + pennies)


# Executes main function
if __name__ == "__main__":
    main()
